src/lucky_cli/wizard/web.cr

```diff
@@ -9,28 +9,35 @@ class LuckyCli::Wizard::Web
   end
 
   def run
+    welcome_to_lucky_banner
     LuckyCli::Generators::Web.run(
       project_name: ask_for_project_name,
       options: options
     )
   end
 
-  private def ask_for_project_name
-    TextQuestion.ask("Project name?").tap do
-      puts "\n"
-    end
+  private def ask_for_project_name : Proc(String)
```

Comment:
I think rather than passing procs around, it may be simpler to extend `TestQuestion` with an optional `validator` proc that can validate the result. It'd go here where we validate that it is present: https://github.com/luckyframework/lucky_cli/blob/db7a0a478fc7cc422b42a011d2e6710f83196db4/src/lucky_cli/wizard/text_question.cr#L14-L19

Maybe the Proc would be `Proc(String, Error?)` where `Error` is a super  simple class that holds an error message. If it returns one then it prints the error message and asks again.

This approach would also mean that  you could use `TextQuestion` with other kinds of validations or later open this up to your own tasks that need validations

Reply:
I'm having a little trouble seeing this. Right now there's 2 types of validations that happen on `TextQuestion`. 1 being `answer.empty?` right there, and the other being done in this `Validators::ProjectName` module. Each have their own validation rule, and error message. It sounds like we'd need a list of validation rules, or a collection of validators and not just a single one. But is this all overkill to abstract all this TextQuestion stuff? We only use it once, and is this something you see people being able to hook in to for use in their tasks?

If this is to be used as an additional tool for tasks, then we should probably re-structure all this anyway to remove it from the `Wizard` namespace. If not, then I'm thinking just throw it all in the class as if it's only used in this one spot any way.... What are your thoughts?

Reply:
I think I get what you’re saying and I agree. Probably doesn’t make sense to abstract validations if it’s only used for the project name. How about renaming it to PorjectNameQuestion and moving the validation there. Like we do with “empty?”

I think that’s what you were saying right? Or something along those lines?

Reply:
Yeah, exactly. Ok, so if it's just going to be used for this one instance, then we can just move things all in to this class, and clean up the logic a bit.

Reply:
Yeah let’s do that. We can abstract later if we want but for now let’s do simple. Good call